packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -147,7 +147,7 @@ export class AmqpConnection {
     this.config = {
       deserializer: (message) => JSON.parse(message.toString()),
       serializer: (value) => Buffer.from(JSON.stringify(value)),
-      logger: config.logger || new Logger(AmqpConnection.name),
+      logger: config?.logger || new Logger(AmqpConnection.name),
```

Comment:
You're saying that the fix is by not conditionally accessing `config`? If so, why doesn't typescript complain here that if config does not exist, logger will eventually crash?

Reply:
TypeScript fails because the constructor enforces saying that a config argument will be passed in

Simply put, `new AmqpConnection()` appears like it should work but it does not

Reply:
@AckerApple So you're saying that the rabbitmq can also be `undefined`? Can we update the constructor param signature to be either the object or undefined by marking it optional explicitly so that way we can control it?
If you want, as part of the constructor you can coalesce to the default instead of having to to these inline's, that is definitely an improvement that I forsee

Reply:
No I'm sorry there is a typing conflict when the first argument is made optional and since I am not a maintainer of this code, I cannot just guess at the proper solution

More about the typing conflict: Because `this.config` is mapped to the first argument, if you/me/we say that the first argument is optional it then creates a chain reaction of typing conflicts. Try and change it for yourself to see more.

Please continue to help this work along. I appreciate you

Reply:
Again this support of no config arguments is needed because of the NestJs Injector code seen below which can result in no config arguments:

```js
        if ((0, shared_utils_1.isNil)(inject) && isInContext) {
            // console.log('instances', instances)
            instanceHost.instance = wrapper.forwardRef
                ? Object.assign(instanceHost.instance, new metatype(...instances))
                : new metatype(...instances); // <--------- no arguments aka config is being sent to new AmqpConnection()
        }
```

Reply:
@AckerApple Review my latest change, ideally, all we have to do is use null coalescence operator `??` operator instead of or `||`

Reply:
I have reviewed your change and that would NOT work because the argument `config` is undefined.

See screen capture where I tried it just for your review:

<img width="2557" alt="image" src="https://github.com/user-attachments/assets/bf81b047-c77e-49f6-bd94-421236c3df1e">

Reply:
I see, let's then add the conditional check that you're describing where you do `config?.logger ?? new Logger` but add a comment explaining why its needed.
It still feels to me that at some place we're lying to typescript and we're injecting undefined, the constructor makes it look like it's required when in reality it isn't ...

Reply:
I'm good for now to simply keep what we've merged and later on, if you'd like, we can propose a refactor that makes this more type-safety and less fragile

Reply:
Thank you humbly for your attentiveness on this subject. Be well friend